contracts/transfer/circuits/src/gadgets/preimage.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details.
+
+use phoenix_core::note::Note;
+use dusk_plonk::constraint_system::ecc::scalar_mul::fixed_base::scalar_mul;
+use dusk_plonk::jubjub::{
+    Fr, AffinePoint, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED,
+};
+use dusk_plonk::prelude::*;
+use poseidon252::sponge::sponge::sponge_hash_gadget;
+use dusk_bls12_381::Scalar;
+use dusk_pki::Ownable;
+use plonk_gadgets::AllocatedScalar;
+
+pub fn input_preimage(composer: &mut StandardComposer, note: &Note) {
+    let value_commitment_x = AllocatedScalar::allocate(composer, note.value_commitment().get_x());
+    let value_commitment_y = AllocatedScalar::allocate(composer, note.value_commitment().get_y());
+
+    let pos = AllocatedScalar::allocate(composer, Scalar::from(note.pos()));
+
+    let pk_r_x = AffinePoint::from(note.stealth_address().pk_r()).get_x();
+    let pk_r_y = AffinePoint::from(note.stealth_address().pk_r()).get_y();
+    // XXX: These need to be uncommented once the `hash` method for the 
+    // phoenix_core::Note is fixed.
+    // let R_x = AffinePoint::from(note.stealth_address().R()).get_x();
+    // let R_y = AffinePoint::from(note.stealth_address().R()).get_y();
```

Comment:
?????

You can already hash this things. 
Also, if that's a gadget as I already told you in our last conversation, you should,t re declare the inputs twice.